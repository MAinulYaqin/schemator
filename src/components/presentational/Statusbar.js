import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { MdZoomIn } from "react-icons/lib/md";

import "typeface-roboto";

const Container = styled.div`
  height: 24px;
  width: 100%;
  background: #b33939;
  display: flex;
  align-items: center;
  padding: 0px 12px;
  box-sizing: border-box;
`;

const ZoomWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-left: auto;
`;

const Text = styled.p`
  color: white;
  font-size: 12px;
  font-family: Roboto;
  margin: 0px;
`;

const Icon = styled(MdZoomIn)`
  color: white;
  font-size: 14px;
  margin-right: 4px;
`;

const Statusbar = ({ projectName, zoom, isProjectModified }) => (
  <Container>
    {projectName && (
      <Text>
        {projectName}
        {isProjectModified && " (Unsaved)"}
      </Text>
    )}
    <ZoomWrapper>
      <Icon />
      <Text>{zoom}%</Text>
    </ZoomWrapper>
  </Container>
);

Statusbar.propTypes = {
  projectName: PropTypes.string,
  zoom: PropTypes.number,
  isProjectModified: PropTypes.bool
};

Statusbar.defaultProps = {
  zoom: 100
};

export default Statusbar;
