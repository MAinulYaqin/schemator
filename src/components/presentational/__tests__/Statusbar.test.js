import React from "react";
import { shallow } from "enzyme";

import Statusbar from "../Statusbar";

describe("Statusbar", () => {
  const setup = propOverrides => {
    const props = {
      zoom: 75,
      pluginName: "Schemator Laravel",
      projectName: "LaravelScheme",
      isProjectModified: false,
      ...propOverrides
    };

    const wrapper = shallow(<Statusbar {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders without opened project", () => {
    const { wrapper } = setup({ projectName: "" });

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders with modified project", () => {
    const { wrapper } = setup({ isProjectModified: true });

    expect(wrapper).toMatchSnapshot();
  });
});
