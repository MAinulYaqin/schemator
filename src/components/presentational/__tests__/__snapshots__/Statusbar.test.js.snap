// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statusbar should renders default 1`] = `
<styled.div>
  <styled.p>
    LaravelScheme
  </styled.p>
  <styled.div>
    <Styled(MdZoomIn) />
    <styled.p>
      75
      %
    </styled.p>
  </styled.div>
</styled.div>
`;

exports[`Statusbar should renders with modified project 1`] = `
<styled.div>
  <styled.p>
    LaravelScheme
     (Unsaved)
  </styled.p>
  <styled.div>
    <Styled(MdZoomIn) />
    <styled.p>
      75
      %
    </styled.p>
  </styled.div>
</styled.div>
`;

exports[`Statusbar should renders without opened project 1`] = `
<styled.div>
  <styled.div>
    <Styled(MdZoomIn) />
    <styled.p>
      75
      %
    </styled.p>
  </styled.div>
</styled.div>
`;
